src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -923,6 +957,835 @@ private FindRequestContext ConstructFindRequestContext(RestRequestContext contex
+            }
+
+            return pkFields;
+        }
+
+        /// <summary>
+        /// Helper method to populate the linking table referencing fields.
+        /// </summary>
+        /// <param name="sqlMetadataProvider">SqlMetadaProvider object for the given database</param>
+        /// <param name="fkDefinition">Foreign Key metadata constructed during engine start-up</param>
+        /// <param name="multipleCreateStructure">Wrapper object assisting with the multiple create operation.</param>
+        /// <param name="computedRelationshipFields">Relationship fields obtained as a result of creation of current or higher level entity item.</param>
+        /// <param name="isLinkingTable">Indicates whether referencing fields are populated for a linking entity.</param>
+        /// <param name="entityName">Name of the entity.</param>
+        private static void PopulateReferencingFields(ISqlMetadataProvider sqlMetadataProvider, MultipleCreateStructure multipleCreateStructure, ForeignKeyDefinition fkDefinition, Dictionary<string, object?>? computedRelationshipFields, bool isLinkingTable, string? entityName = null)
+        {
+            if (computedRelationshipFields is null)
+            {
+                return;
+            }
+
+            for (int i = 0; i < fkDefinition.ReferencingColumns.Count; i++)
+            {
+                string referencingColumnName = fkDefinition.ReferencingColumns[i];
+                string referencedColumnName = fkDefinition.ReferencedColumns[i];
+                string exposedReferencedColumnName;
+                if (isLinkingTable)
+                {
+                    multipleCreateStructure.LinkingTableParams![referencingColumnName] = computedRelationshipFields[referencedColumnName];
+                }
+                else
+                {
+                    if (entityName is not null
+                        && sqlMetadataProvider.TryGetExposedColumnName(entityName, referencedColumnName, out string? exposedColumnName)
+                        && !string.IsNullOrWhiteSpace(exposedColumnName))
+                    {
+                        exposedReferencedColumnName = exposedColumnName;
+                    }
+                    else
+                    {
+                        exposedReferencedColumnName = referencedColumnName;
+                    }
+
+                    multipleCreateStructure.CurrentEntityParams![referencingColumnName] = computedRelationshipFields[exposedReferencedColumnName];
+                }
+            }
+        }
+
+        /// <summary>
+        /// Helper method to get the name of the related entity for a given relationship name.
+        /// </summary>
+        /// <param name="entity">Entity object</param>
+        /// <param name="relationshipName">Name of the relationship</param>
+        /// <returns>Name of the related entity</returns>
+        public static string GetRelatedEntityNameInRelationship(Entity entity, string relationshipName)
```

Comment:
This helper method does not fit in here. Maybe move it to `GraphQLUtils` since relationships are only relevant for GraphQL?

Reply:
Moved to `GraphQLUtils` class